### ISSUE TITLE
Support enum.auto()

### ISSUE THREAD

**Issue by rmunoz12** [python/mypy#2609]
This fails, python 3.6:

```python
import enum

class Bearing(enum.Enum):
    NORTH = enum.auto()
    SOUTH = enum.auto()
    EAST = enum.auto()
    WEST = enum.auto()
```

Running mypy gives the following:

```
01.py: note: In class "Bearing":
01.py:10: error: "module" has no attribute "auto"
```

**Comment by JelleZijlstra** [comment]
That sounds like an issue with typeshed; https://github.com/python/typeshed/blob/master/stdlib/3.4/enum.pyi is missing this function. You should report this as a bug (and/or provide a patch) to the typeshed project.

**Comment by gvanrossum** [comment]
I think the fix hasn't been merged yet?

**Comment by gvanrossum** [comment]
Whoops, sorry. The GitHub message mentioned a commit in Michael's fork of the typeshed repo and I didn't double-check the history carefully.
